{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#This code identifies which genes linked to melanin synthesis contain a Pannier \n",
    "#binding site within a 2000 bp region upstream of their transcription start sites."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Find 1:\n",
      "Pattern_1 (GATG) found at index 213\n",
      "Pattern_2 (TGATAA) found at index 218\n",
      "Number of nucleotides between the patterns: 1\n",
      "\n",
      "Find 2:\n",
      "Pattern_1 (GATG) found at index 960\n",
      "Pattern_2 (AGATAA) found at index 971\n",
      "Number of nucleotides between the patterns: 7\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#WERK1 Henna\n",
    "#Gene ID: 123682245\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    # loop:finding p1\n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            # if p1 found, p2 (400bp)\n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    # if p2 found,cal dist\n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "#test sequence\n",
    "genome_Seq = \"\"\"ATAACAGATAAAGCTTTAATTCTTGATATTATTTCGAAGATCAGGGACGGCTCACCCTGCGATATGATCG\n",
    "CGACCAGAAGACTCGGTAATTTGCGACAGGGGGCGCCCAGACTGGTTAAGGTAGAGTTTTCGAACCCGGC\n",
    "GGTACCTGCCAATATCCTACGTCATAGACGGGTCCTTGCTGCCTTTGAGAACTACAGGAGAATTTCAATA\n",
    "GATGATGATAAGACGCCACTTCAAATTAAGCAGCTTCAAGACCTTCGCAAGGAGCTTGAAAGAAAACGTT\n",
    "CTTCTGGGGAGACTGACCTGACTATTAAATACAGGAAGGGAATCCCTTCGATAATCAAAATACCGTCCTC\n",
    "AAAAAATTGAATTCTAACTTGATCACGTTGCCACTCATTTATACGAATATTGTTTCTATTAACTCAAAAT\n",
    "TCGAACAGTTATCTGTCTTCGCTTGTGATCGTGTTGGAATCAACTAACGCATGATTTTGTTGGAGCTGCG\n",
    "TCGGTGAGCGTATTCAAGAATGGGTTCGACATCTGGTGGACAAGAGTTCTTGAATGAAGTCAAATATATT\n",
    "TTCAAATATCGGATGTTTCAGTATGTGTTGTTTTTCCTGAATTGTTTTGGTGTTATTTTTTCACATTAGG\n",
    "TTTTATTTTGTTAAGAGTTTTATAGTTTTATACTCAACTCTGTATCCAATGTAATAATAATAATAATAAT\n",
    "AATAATAATACAAGTGCAGAAGGCATTGATATTCTTCCACGAGTTCAAAATTCAAAAACGAGCCACGAAG\n",
    "TGGCGAGTTTTGGAATGAACGAGTGGTAGAATGAGCCTTCTGTACGAGTATTATACATTATTTTCTCTAA\n",
    "TTCATTGCATTTTCATTGAAATCAATGAAATATTTCCATAAATATAATTTAGTGATTTTTGCATTGAAAA\n",
    "ATGTTGGTTGGCAGAACTGATTTCTTTAAGGCAGATTGATGAATTGACAGATAAAGCCATTGCGGAAAGT\n",
    "TCGGAGTGCCAACATAGAATAATAAAATATAACCATAAAAACAGGGCGTTTCTGATATATTCTCGCACGA\n",
    "TTTTGTTCTACAAGATGTGGAAGAATGAACGGAATAACCACAGAATTAGAGAAAATACTAACTACTATAT\n",
    "TACTGAAAATAATGGTTTTGTAGTAAGTACCTTCGTGTCGTGCATTTCGGATTGATTATGGATAATCGTC\n",
    "AATCTGAGCGTAGAATTGTGGAAACAGTGGAAACATTCGAAAATACTGGATCTGTTATTGATGTCTTAAG\n",
    "GCCCATACATTATCGAAATGCACGCTCGGCCAAAAATGTTGCTGCTGTAAATGAAAGTGTCGAAGAAGAT\n",
    "ACAAAATTATTTATTCCTCGGCGCGCACAACATTTGGGCCTCATTTACGGTACGCTTTGGCGTATTTTGC\n",
    "ATCTGGCTTTTCATCCACCTCATAAGGATAACGTTCAGTTGACACAAGAACTAAAGCCAGCTCACAATGG\n",
    "AATGCGTAGAACATATTTTGATTGGATGCTCGAACATTTAGAAGACGTCGAATATAGAGCATTGAAAATC\n",
    "CTGTATGGTGTGCAATTTGGCCTAGTGGAGTGATTGGTCCGTACTTTTTTGAAAATGAAGAATTCAATTC\n",
    "TCATTGCTGTAGCCTAATAATTACCAATTTTTTGTGGCCCGAAATCAATGATATGGAATTAAAAGAGATG\n",
    "TGGTTTCAGCAAGATGGTGCAACAAATCACACGGCCCATTTTAGCTTTATTGAGAAAAAATCGTAACAGG\n",
    "CTTAATTTCTCATTTTGTCGATGTGTGTTGGCCACCAAGATCTTGTGATTTATCGCCCTTGAATTTTTTT\n",
    "TTGAAGGATTGTTTTTATGCCGATAAGATCATCCACAGACTATTGACGATTTGAAAATCAACAACCAACA\n",
    "TCTGTCAAGTTATTGGTTAGATATTGCTCGAGATCTACTGGCGAGTCATCAAAAATCTCTACATGAGAAT\n",
    "TTAGTGCTGTAAGAGAGCACGCGGAGGACATTTAAATGATG\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No occurrences of the patterns were found.\n"
     ]
    }
   ],
   "source": [
    "#WERK2 TH (ple)\n",
    "#Gene ID: 123673805\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"ATTTATGTTTTTCAATTTTTTACTAAATTTGCTCCTTATAGCATATGCAAATTCAAAACAATTTGATTAC\n",
    "AAAAGGTTTGTCCAAGTTTCCTGCAATGCATTAGGATCCAGGGTATTTATTGCTTAACAGTACAAACAAT\n",
    "TGAAACCTGGTAAATCATCTCTGGAACCAGAATGTGTAGACTCGGTTGTTTCTATTGGCTAATATATTCC\n",
    "ACGTTCCACCGCATGCCAATTTGGTTACTTTCATTGGCTGCAGCTGTAACTTCTTATCATATTTTTTTGA\n",
    "TGATTGAATATACCTAGTGTGATGCGTAGGTGGATTAAGGACATGAAATGAGTATATTGACTCGGCTTTA\n",
    "GTTCGATGTGTAGCTGATTTCTTCTTTTTGGATAAAGAATATTCAAACTTGAATAGGGCAGGATATTTCT\n",
    "ATTATTCCAAGTTACCAGAAATTCATTATCAGGCAGGGACAGCAGAGAATTTATTATCTTACCAGTCCTT\n",
    "TCGAATAGATTCCTGTATGCTAGTTAATCGTCATATGAAATAGTATTTGTCGATTTGGTTACTTTCATTG\n",
    "CCCGAAATATTAGGTGTACAACTTTACTTCCGCGTTTTTGCAATTGATGGCTGTAGCAGTTGATGGTAGT\n",
    "CGAAATAAATAGATCGTAGATGTCATACAATAATCTAAGTTATTTGTGAACATAACGCCAACGAAACATT\n",
    "AATCGATTTCTGTCTGCATAATGAAGTTATTCTCAATTGAACATGTCAGCTTACGAGCCAAATTCTCGTC\n",
    "ATTTGTGGAAGGTTTTGATTTTCTCCTTTAATATGAAAAAATCTGCGGCCGAGGCTCATCGAATGCTCTC\n",
    "AAATGCCTATAGTGAGGCTGCTATTAGTGAAAGAACGTGCCGAGAGTGAAATCAACGCTTCAAGAACGGT\n",
    "GATTTAGACGTCAAAGACCAGCATGGCGGAGGAAGAGAGAAGGTTTTCGAAGAACGAAGAATTGTAGGCA\n",
    "TTACTTGATCAAGACTCGTGTCAAACTCAACAAGAATTGGCAGGATCATTGGGAGTGACACAACAAGTAA\n",
    "TTTCAAACAGAGAGAGATGTTGAACGACGTTTGTTTGATTGTGAACTGCTAGCAGAACAAAGACGGAAGG\n",
    "GATTTCTGCATCGCATTAGGACTGGAGACAAATGAGCTCATTACGATAATCCCAGGGCAGAAAATCATGG\n",
    "GCATGCTTCCACGTCGATGGCCAAACCGAATATTCACGGTTCCAAGGTCATGATCAGTATTTTGTGGGAC\n",
    "CAGATCGGCTTAGTGCATTATGAAATTTTGAAATCGACTGAAACAATCACAGGCGATCGTTATCAAGCGC\n",
    "AATCAACGCGTTAGAACCGAGCATTGACAGAAAAACGACCACAATACAACGAGAGACACGATAAAGTGAT\n",
    "TTTACTGCATGACAATGCTCGACCCCATGTTGCGAAAGTGGTCAACAGTCAAGACATACTTGGAAACGTT\n",
    "GAAATGGAAAGTCCTACCTCACGCAACGTATTCTCCAGATGTTGCTCCCTCGGATTATCATTTGTTTCGA\n",
    "TCAATGTCTCACAGCCTGGCTGACCAGCATTTCCGGTATTATGAAGAAGTAAAAAATTTCGATCAGATCA\n",
    "ATTTTTTAAGACTTTGTTATTTTATTAATGTTCAATTTGTAGGGAATCTATCGAGATTCAATATCGAATA\n",
    "TTCTTTTCAGAATCAATAGAAGACAGGAAGGGAAAAAATATCAAATTTTCACAAAAGTTGAACTTTTAAT\n",
    "TTAAAAAATTTTTCATTAATAATTCTCACATTAATTCACATAAAATGAATTCTCATCAAAAATTTTATAT\n",
    "CTTACACTCAAATTATGGCCATTCAATTTTCAGTTTAAACTTATTTAACATATGATAAATTACACCATCA\n",
    "CAGTATTACGTGAAATTTTCAGTCACATTAAGGAGTTCCTGTTTAGTTATAAATAGATGGAACTTCAATT\n",
    "AATAGAAACAGTTCTGGTAACTTCATAAACATTGAAATATC\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Find 1:\n",
      "Pattern_1 (GATG) found at index 1389\n",
      "Pattern_2 (AGATAA) found at index 1397\n",
      "Number of nucleotides between the patterns: 4\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#WERK3 DDC\n",
    "#Gene ID: 123679005\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"AGTGGACAACAAACATTCATAAATATAGAGACCATATACAGTCATGATAGGCAAGGTCTTAAATTTGCCT\n",
    "CTGCACGATTTTTTAAATTTCATGTTATATATCGTCGGCGTAAGTAAGTGTTATAATAATAATAACAGTA\n",
    "AGTATTATAATAATAATAATAATAAGAGAGTTTATTCATGAAAATACATTCAATAAACTCTATTGAATGT\n",
    "ATTTTCATGAATAAACTCTCTTATTATTATTATTATAACACTTACTGTACGGTTTTTTTGGTCTCCTGTA\n",
    "AAATTGTTCATTATGGACATAGCAGATTTACACTCTTAGCCGACGATATTGTTCTTAGAGTTCTTTTCTG\n",
    "CACAATGAAAACTGACTGAATGTTAGAGTTGCTTCCCCAAAAAATTATACCATACTTCAGATTTGACTCG\n",
    "AAGTTAGCACAATACATAATTTTAAGCGTATTTATATTCATATAATTAGCGACAGTTTTGATTCCGTATC\n",
    "CGATACTTCCCAATTTCAAGCACAATTGACTTATATGCGTTTCCCAATTTAAAAATTCATCAATATTCAC\n",
    "CCCCAAGAATTTTGTAGATGTTGCTGGATATATTGTATGCTTTTCGATTATTTCTGGATTGGACAATTTC\n",
    "GATTGTTTGGTTCCGAATAGCATTAAATTGGTTTTAGATTCATTAATTTTGGGCTTATTCATATCGAACC\n",
    "AACATTTTGCGTCAACAAAGATGATTTCTGGTTTTTCGAGAACTTCTGGTAAATTTTCCCGCCCACTAAT\n",
    "ATATTAGTGTCTTCGACATAATTCGTCGTAAGGCAACAATCTAATCATTAAGGAAGGCCATTCATAAAAA\n",
    "CTATGAAGAAAATGGGTCTAGCTATATTCCCCTGTGCTATACCGGTATTGACAATTTTAATATCAGATTT\n",
    "AGTAGACTTTTCAACCGAAACCTGCTGCTTTTTATTCGTGAGAAAAGAACGTATCCATTCATTCGCTCTT\n",
    "CCTCTTATGCCATATTTAGGTATCTAATTTCAGTAGTAGAAGCTCATGGTTAACGCAATCGAAAGCTTTG\n",
    "CTAAGATCAATGAAAAGACCTAAAGCTAAATTTTTGTTCTCTAAGTGATTCAAAATTGCATTTATGAACT\n",
    "GATATATGGAGGTATTCGTTGATTTTCCTTTCAGATAGCCATGTTGGAATTTGGAGAAAATAATGTGTTG\n",
    "ATTCATAAAATTGAGCAGTCTAGAAATCATCACCATTTCGAAGATTTTTGAAAAAGCAGGCAATAAACTA\n",
    "ATTGGTCTGTAATTGCCCAGATCATCAGGATCCCCCTTTTTGAATATTGGTTTGATAACAGTAAATTTCA\n",
    "ATTGATCGGGAAAAATTGAATATTTCATCGAGTTATTAATGATGTAACAGATAATTTTTTTATTGATGAT\n",
    "TTCACGATGGCAATAGGTACGTCGTTATAACCAGAACAAAGTTTATTTTTTCTGATGATTTGACGAACCT\n",
    "ACAAGTTCAGGTGGGGTGACCGGTTTAAGAAATATCGAGCTATTATTTTCTTGAATCATACATTCGAAAG\n",
    "GGATATTGCTGATAATTTTCGAAGAGTTAAATGGCGTGCGAGACACTAAAACCTACTGATCCCCATTGAA\n",
    "AATTTTCTATATTCCGATAAAATTGCAATATTCCACTAAACAGGGTGTTGAATCTCCCATAATTCCCATA\n",
    "ATTTTTTTAATACGGACGCTTTACTCATTTCCATTCGATATTTGTATACTTAACTTTCAAAAACGAAAAT\n",
    "TGAAGATTTCACAGGGGGCCAAAATTTTTTTTTCGAACGATTCACAGCTTCATTTTATTGGATACTTTCT\n",
    "TTCCGGTATTCCGGTGATTGATTCATACATAGTACACTACTGAGGTTACGACCGTCTATCTCTCATAATT\n",
    "GGAAGTTGGCAAGCTGCCTCTAATTACCACATTCCACTTATTGTCATCAACAGTGATTTCCCATCGACAT\n",
    "GACGTATGTAAATATCAGTTCGGACCTGAAAACCCAGTGAA\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No occurrences of the patterns were found.\n"
     ]
    }
   ],
   "source": [
    "#WERK4 tan\n",
    "#Gene ID: 123674268\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"CAATGCAAAAATCACTGAAATATATTTATTGAAATATTTCATTAATTTCAATGAAAATGCAATGAATTAG\n",
    "AGAAAATAATGTTTAACTCGCCACTTCGTGGCTCGTTTTTGAATTTTGAACTCGTGGAAGAATATCAATG\n",
    "CCTTCTGCACTTGTATTATAAATAACTATACTGAACGTACCTCAATTGTTTTTCAGTTTTCTACCTCTCC\n",
    "TAGAGACGGGGTAACCTTTTTTGGAAACACCGTGTACACAAGGTAGTCCAGATTTTAGTGCAATAACGTT\n",
    "GGCGTCGGATATAACAACTATTTTCATGGAAAAAGTTCATATGAACATATATCCTATCAAACGTTTCCGA\n",
    "GATACAGGGTGTTAAAATTAAAAAAAAAAGTTTTTTACTTATGGCTCTCTCTCTATAACTATTATTCATT\n",
    "TTTTTAAATTTTTAGGTATGAAAGTCTAGACAATGTAATATTTCGAATTTGGCAAGTATCTCTTGCCAAA\n",
    "ATTATCCAGTGGCGTAAATATAGGGGTGCGATTATCCATTTCCTATTGAAAGTCATCATCCCTTTTTTTT\n",
    "CGACAAAATTATTCCTTTTTGAAATCAACAGAGCTTCACTAAAAAAAGGTCCCTTGAATTTATTTCATAG\n",
    "AATGTACCATTTTCGAGATGATCTAATACAAAGCAAACATTATTCACGAAAATCTACAAATATTAATTTT\n",
    "TCTAATTTTCTCTTTAGTGATCAATGAAAGTAATAAAATTGACTTAGTGTATTATAATACTTTAATTGAA\n",
    "CATTTCATAAAGAACATTTATCAGACATTCAAGCACCAAACTATTTTTCTGATGACGCATCGGAACTTTA\n",
    "AATGTATACGTTCTGGTTTCAATCCAAAATTCAGAAAATCAATCAATGGAAATATCTCCAGAAATGCGAT\n",
    "AAATGAAAATTTTACTGAAAGTAGTGTTTTAGAGGAAGTGGTTATTCCATGTGATAAAAAAAATTTCGAA\n",
    "ATTCATCATTTAAATGAGAAATATTTCAATTTTCAAAGTAATATAAAAAAACCATAAATTTTTTTGTGAA\n",
    "GAAATCAGTGAGGTTATTATAAATTGCCAAACGAAAGTTACTTCCAAAAAGTGAAAAAATCAATAAAGCA\n",
    "CAGAATCAAAATGAATCTCCACTTAAGAGACTTAGTGGGAAAGATAGTAAACATGCGGTATTTATTAAGT\n",
    "TTCGGGCCTACACTGAAAATTTCAGGTTTCTAGGTCCGTCCGTTCGAAAGTTATCTTGTATACAGACGTA\n",
    "CAGAATCGAATTTTTGGACGGATTCATCATCAGTGGGTCAAAGCTTCTCGTTTGAGTAGTAGTTATAATT\n",
    "GATTGGAGCTGAAACCAGAATCTGTCCCTAACTTTGTAAAGAAATATAGTATTAGAAGGTCAGGCGGATC\n",
    "GATTAGTGACCATTTAAACTTTTTAAAAACAAGCGCAATATCATTCAGTGAGAAAAGAACTTAAAAATTA\n",
    "GTTGATCTTAAATCCGTATAAAATAATTAACCCCGAAACTATTGAACCTTAAACATTCTGATCGAAACTC\n",
    "TTTCCTTTGAATAGCTCATAAAATTTGTTTCGATATACACATTGTACACATTAGGAAATGTAAAAAAAAA\n",
    "TTTAAATGTGAATTCTCGAAAATTCCTCAGATTCGTCTGATACTCATTGAACAAATTCTACCCATAAGCC\n",
    "GCTCACGTCATATATCCACATACATACATATATCCAGTTTATTCAAAAGAGTTTCGCTATGAGTTGAAAC\n",
    "GTAGTAACCATTAAATAATCCCAACCTGAAAATAATACAGCAAAAACATTACTTTGATAAACCCAAAATT\n",
    "CTAAAATAAACATAAACTGTAGATTCTCATGCTCCAAATCCTATCAATTTACTTAGACCCACATTTTTAT\n",
    "GGTTTCATATCATTGTCAACTAATTAAATAATACCTTCTTTAAATTTTATCAGTTAACGATAAAAAGGGT\n",
    "GGGAGAAGTTGCGCGAACTTGAATTTGCCATAGAAAGGTTT\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No occurrences of the patterns were found.\n"
     ]
    }
   ],
   "source": [
    "#WERK5 yellow / y \n",
    "#Gene ID: 123676132\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"AAATAAAGATTTCACATCGAAAACAATCCTTCACATGAATTCGAATTGGTCCAAATTATTTCCCTAAAGT\n",
    "GAAAGTGAATGAATTTCAATGAACCAGATGAATATAGTGTTTACAATTTTTTGGGAGAGAGTTCATTTTT\n",
    "TCTACTTTGGTCACAATTACCGAAAATGCAATAGCTTTTATTGGCTTAATGAATTCTCTAATTGAATTTA\n",
    "CGACCCGTTAAGTTATTCGATAAGATTTCTTGAATTGGCATGAGAATGCGGGATTTCAGAATTAAAGAAT\n",
    "CCATAAATTATCAACTTGATGTTTTGAGTGAGTAACGAAGTGTTGATATTTCAACACCTTCAACAATAAC\n",
    "TTGTTTCAGTTTTAAATTTGTTTTAGAGAATTATAACGAGATAAAAACAATAGATCTACTCAATAGATAA\n",
    "GTTTGGAATGAAGACGAATTTGCGAAGAGGTTAATATAATTTGTATTAAACAAAATAATACCAAAGAGCG\n",
    "CAATATTTTGATACAGGTAGATGAATAGGTGACTTATAGGAGTTAGTAAACATGAAGTGTTACACTATAA\n",
    "ACTACAAAAAAAAATAAAAAAAAAATGATAAACTACTACCACGAATAGGATATCAAAACAATAAAAAATA\n",
    "TATGCCGATTTTGAGCGTAGAGTAGTTTTTAGAGATGAATCTTCATAACTTAATTGAAAGAGAGTCCAAG\n",
    "TTGGCTTTATTTCATAAAGAGAATAACTTCTTCTGCATATTTCTCAATCTTTTGTGTATTTGATTATAAA\n",
    "TAAAAATTTAGAAATATTTTACATAATACTAATAATCTTATAAGTGGCCTGATTGGATCGAATTTGACAA\n",
    "ATGATATTTTTGATTGAATTGGGATCAAGTATGTGCCAATTTAAAAAAGTACCATCCTAAATATCTGGAT\n",
    "CCTCATTCAAAATCAGAAAAAATGCAAAAACAGGTAAACTTTGATTTGTAGGGGGATATTTTATACAGTT\n",
    "GGTTCCAATCAATTTGTATCAACCTTGTGAGCGTAATGATAGCAACCCCTAAAATCTGAAATTGGAAGAG\n",
    "GACACCTTATTTTGAAGGCCTTTCGTTTGCCTTTTCATAAATGCCATATACTTGACATCTTATTGCGAGA\n",
    "AAAATAAAAAAAAGAAGAATACTTAGTTACGATAGCAAGAGTTGCAATGAAACTTGAAAATAATAAAAAA\n",
    "TGTTTTACTATAAAAGATTTTCAAAATATTGTCCATTATTTTATAGACAAAATATTAATAATTTTCAAAT\n",
    "TCGGCCTTTACGATATGGAACATTTCAACTTGATCTTCTCGGCATGCTGTGCCTATTTTTACTTTCAATT\n",
    "ATTCTACTTCTCGGGGCTGTGTTTTCTGCACAACCGACTTTGAATGTCACCAGAGGAAAAAATCTGAAGG\n",
    "TGTCAAGTCTGGCTAGCTTGCTGGCCATTCAATTGAAAAAATCTAGTAGGTGGGTGTCACGTTTTTTCAT\n",
    "TGAGAACATGATCATAAAATTCCATTCCTTCGATTCTTCAGCAATTTCAAGAAAAAAAAATTGAAATTGT\n",
    "TTATGTTTTCAAGGAAAGGCGTAGAATCTTTTGTTTACATGAAAGACCCCCTTTAATTTTCGATAAGGAT\n",
    "TGATCCTTTGAAAACTTACTCATTTGTAATACTGACCCTATATTGTCACATTATACGAACACATCAAAAA\n",
    "ATTAATAATTCTAAACACCAATTTGCGGAATTGTTATTGATCCATTCCTTCCATAATTGGCAATAACTCT\n",
    "CGTCGCCAGAACCAGTTGAACCTGCGTTTACCACATTTTCGGAATATTCAGTAGTATATAAAAAACACCC\n",
    "ATCTAACGAAATGCAAATATCTCTAAGAGTTGAAAAAACTCGGAAGACTCTCAAATCTACCAGAATTTCA\n",
    "TCCAATTAATTTCACGAAAAAGTGATCGTCAACCGCCTTGGTCAGCAAGGTAAGATTGTAATCCCCCCTC\n",
    "CCCGTGTTTTTTATCCGATTCAATCGAGTGAGGGTATTTCA\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No occurrences of the patterns were found.\n"
     ]
    }
   ],
   "source": [
    "#WERK7 laccase\n",
    "#Gene ID: 123676796\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"CAATGCAAGGTCATTGACCTAATAATGAGTTGCCTTTGGTAAAAAAGCTGTAAGTGTGCAAATGGTTTAA\n",
    "TTTTAATATATTTGAATTTGGTAAATTGTTTGGCAGTTGAGGCGTATCGTGTGATTTCTTGTAGTACATA\n",
    "AATTCTCATCATTATACATCTTCTTTCTGTATAAAATAATGATTTCGACTCAATTAACTTCAAATTTTCA\n",
    "GTATTATAGTTATAATATTTACATCTTTTTTATATTCTTGTTTTATTTCATATATATTTCGTAATACTGA\n",
    "AATCTGAGGAGAGTAGAAAAAACTTACCTTACATTTTATGAAAATTACTTCAAGAATCATCAGAGCTTTT\n",
    "CATTTTTTTGTACATTATTAAACCAAAGGATTCAAACATAGACTTCTAAACTTTGTGAAGACTGAAATTA\n",
    "TCAAAAGCCCTAGCAGCGACAACTATTAAATTTGAAAGAAAAAAGTACAAACTTCAATTATTTTCTTCCA\n",
    "TTGTTCAACTTCTTGTTATCTAAGTATATCGAGAAAAAGGCATTTAAAATAAGATATAGCATGGAATGTG\n",
    "TGTAATATTTCTGGGCATCTTTATTAAATTTTGTTCATATTGGATATGAATGATGAATGACAAGATATTT\n",
    "GTAACTTTTCTATAGTGTGATATTTCCGTTTTGTAGTTGTGATATACTTATATTATAACCGCACAAATTT\n",
    "ATTATCTTCCATTAACTGTAATGTAATATGCACACAGGAATCGTTCTTCCAAATTATTTTAAGGAATCTA\n",
    "TTATGTGTTAATATTAAACTGTAAAACCAAAAATTGTGATAAAATTATGTGATGTAAAATGTGAATTTTG\n",
    "AATTTATTCATTGGATTGTGTTCCATATAAGATACACAAGGTAAACTTTGTTTTAATGAATATTCTCTTT\n",
    "GATTCATTGGTTGTATCCAGTGAATAAATATTTGTCTCTCAGATTGATACGAGCCATATCTTCGTTTAAC\n",
    "AATGTAATATGAAATTTTATATTTAATTATAGCTTAAATTTATCTGAGAAACAAAATTTTTTTCATTCTG\n",
    "GATACTACCAATTAATCAATTTTGATTTATTGAAATTAAACATTGTGTACTTTGTAGTTGAGTATTTAAT\n",
    "AACCTTTATTATGACAAGATGAAGTGATTCAATGTTCCGATTTGTGACTAGGAAGTTTTTAATTTTTTGA\n",
    "TATTGTGGAATTAAATTTTTTCGCGATGCTATTGTGATTCAATAAAAGTTCATTTGTCTCATAAATCTTG\n",
    "TTTAATTTACAAAATAACGTCTTACAATTAAGTCAACTATAATGAACTACATTTACATACCAGAAATTTG\n",
    "GTTAAGAAAATCAGTTTCTTGAAAGTAGTGAACTTTCTAGTTCTTATGATCAATGAACTCTGAAGGGAAA\n",
    "AAAGTACTGATTAATTTTTATTTGATGATGCACCAAATTGGATAATTCTTTTTCTGTTGTGTTCATTACT\n",
    "GTCATGTTGTTTGTACAACGAAGTATTTTCAAACAAATCGTACGGAAAACAAAAAAAAGGTTTTCCTTTT\n",
    "TCCTCCAACCAAGTGAACAATTATAGAGTATAATTTGACACTTCGAAGAAAGTATTTTGGGTTGCCTGAC\n",
    "AGAACGTGCGAAGCCGGAGATGGCTGCTAGTTTTGAATAAAAAAACTCCGTTCAAATTTTTTTTTCAAGT\n",
    "TGTTTATGATATTAGGACACAGAAAAATTGGAAAAATATCTAACTGAATATTCTGTACAAAATAAGGCCA\n",
    "TAAACTAATGAAATTGGAAAAACAGAAGATTAATTTTGAAACGAATGTTTATCATCTTAAATTCGTTAAT\n",
    "CCAATTAAAATAGGTACCTCAATTTATATTAATATAAAGAACAATAAATAAAGGTCCCCATATATTACAT\n",
    "TTTTCGGTTCGGATTAGCTTAATACTAATTTTATTGTAGGATCAAAAGATAAAATGAAATCCCAAATGGA\n",
    "AGATTCCAGCATCCTATTCCATGTTAATGATGTTTTTAAAT\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No occurrences of the patterns were found.\n"
     ]
    }
   ],
   "source": [
    "#WERK8 GTP cyclohyrolase\n",
    "#Gene ID: 123679556\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"AGAGAAATAAGAAAATTAAAAAAGTTCAAGAAAGAATTTAAAAACTGTTTCATTAAAGAAGATCGAACAT\n",
    "TTATAGATTTATAATCAGTCTAATCAACTCCATTGGCGACGCCAGCTTTCAAAAACAGCGGTGGCCAAGA\n",
    "AGGGACCGGATTTTTTTGGGGGGTCAAAGTAAAGCGGAACTATAAATCTGCTAATATTTTAGCCTTAGTA\n",
    "TGTCAAATTTTAGGGGGGCCAGCAGAATTTTAGGCCTCCTCCCCTATGTTCGCCACTTCATGAATTGAGT\n",
    "TGGTTTTTTAAAAGATGAGATGATGAAAGAAAATTCCCGATTATGAGAAAAATTTCCCAAATGCGACCAA\n",
    "TGATGATTATGGAATAATGGTGGAAAGAACCGTATTTTATTTTTCTGCTGTTGGAATCATATTTTTATTT\n",
    "TATAGCTGTAATCAGTTAATTTAGAGATAACTCACAGTTTATTTTCTGAGTTTTATTTGAACTTGTACCT\n",
    "ACTTGAATGAACTCATTGATTATTATTATTACTAATGCCGAATGAATGACTAGAGCTCAAAAGCTTTTGG\n",
    "TTTCACGTCTGTGGTTTCCTCTTTTTTTTAGAAACTTTCTCGATTTCATTCTCCTTGAATATTCTTCGGT\n",
    "TCTGAAAATTCTTCTTGAATTTTTGAATTTTTCCTTCTGTTCTGAAAGTGTAATCAACTTGAAATAGCTT\n",
    "TCTCTCTGGCCCTGGTGACGCGATCTACACAATTTTTTTCAAATTGGTATTTTATATATAGGGTGTCCAT\n",
    "AATTTACGCGACGCTCTATTACGGTTGAGCGCTGAAATTTTCAATTTTTTTTCGACATGGTACATAGGTA\n",
    "CCATTGAGATACATTAGAACTACATTCTCAAATTGTACAAGGGTGTACCAAAAAAGTATGAATAAATTTT\n",
    "TTTTCTTATGAAACACCGTGTTTTATTGAATTTTCAGATAACATGGAATAAATGAACCCAAGTATGTTTG\n",
    "AAGTTTCATATGCTTAAAATTGTCCGTTTTTGAGATAATTCGAATTCATTTGAATTATATTTTGGAAATG\n",
    "AATACCCACAGAACGGAGTTGAATATTGATTTTTGATAATCATTGCGAACTCTACTGGGTGTTTGAAATG\n",
    "AGACCATTCGTTTATGATTTGTTCGAACGTTAATACCGGCCTCATTCCAAAAGACACATCCGACATTTTT\n",
    "TATTGTTATTCTCATTTGGTAAACAATTTTTCGAAGATAATAAAATTTACTGATTAATGACAAAATCGAG\n",
    "GGTGTGCCATTTGGAATAAGCCCGATATAAACGTTTGAATCGTCATAATACTCGTGATAGAGCATCATAA\n",
    "TGAAACACTCTGTATAAAAAATGAGGTCTATGATAATTTGTTTAACGAATATAAAGCCATGAATTTCACT\n",
    "TGATCATTGCAAAAAGATGAATATCTACACTTCCAAAAAGTTCTTACTAGCTGCAATATTGACATAATAT\n",
    "CCCGTAATTGTTAGGTATATAGAAACTGTAATTGAGTCAAACCTGATTCAGATATTATGACATAATCCAA\n",
    "TTTAAACACATTAATTGCACATATGAATCAAATTACATGAATGAAATTCCAAATGCTTGTTACTTATTTA\n",
    "CTTATGTGATAATGAAAGAATGGCTTTCAAGATCAGTTAATGCGATTGATTTCACATTATTCACATCTAC\n",
    "AAAGCGAACGATAAAAAGTCCTGCACTAAAAAAAATCGTGCATATTATCATTGTTACTTTTTGTCGATTG\n",
    "AAACCACTGACAAAACCTTGGCCTTGTATCGTACGTAGTCAATTAAAACAATTATTCAATAATTTCATGC\n",
    "CCCTACCACTTTTCAAGTGGATTTGATCCGTAGATAATACCTTTTTCCCACCTTTTTTATCTTATACGTG\n",
    "TGATTATTGAATTTGTGCAAATTTCTTTGAGTGTTCTCAAAACTGTGACTTTGAAATCAATATCATGGTT\n",
    "TATCTTTTTCCTATCCGTATCAATCATCATGAAGTTGATTT\n",
    "\n",
    "\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No occurrences of the patterns were found.\n"
     ]
    }
   ],
   "source": [
    "#WERK9 ebony\n",
    "#Gene ID: 123680340\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"ATCTAACGTGTTTAGAATGAAGACAAAACAAATAAAAAAACAATAAACAACATCTTTGTTGTCATCTTCG\n",
    "AACATATACAAAATGCCTAATACTGGGTAGTTATTGCACGAAAACCAAATTCTTGATTTTTCACACACAA\n",
    "AAATCACATACAAACACAAAAATTCGTATCAGAAACATATGATGAGTGAGGAGGTACCCGGGTAGCAGAT\n",
    "CTAATATAGGCAACCAATCATACACTAAGACGTTTTGTGTGTTGCCTATTCAATTGTACTCCTGCTCCTG\n",
    "GTATACCTCCTCCATCGTGAGATCGAGAACATATGAATGTTCCTGTTACAAAATTCGTACACCAAATTCT\n",
    "TGTTGTTTCTCGTTCGAATTTACTCATTTCACACACGAAAATCACATCCAAACACAAAAATTCGTATCAG\n",
    "AAACATATTATGAATGAGGAGGTATCCGGGTAGCAGATCTAATATAGGCAACCAATTATACACTAAGACG\n",
    "TTTTGTGGGTTGCCTATTCAATTTTACTCCTGCTCCTGATATACCTCCTCGATCTCATATGTTCCTGTTA\n",
    "CAAAATTCGTACACCAAATTCTTGTTGTTTCGTTCGAATTTACTCATTTCACACACGAAAATCACATCCT\n",
    "AACACAAAAATTCGTATCAGGAACATATAATGAGTGAGGAGGTATCCGGGTAGCAGATCTAATAATAGGC\n",
    "AACCAATCATACACTAAGACGTTTTGTGGGTTGCCTATTCAATTTTACTCCTGCTCCTGATATACCTCCT\n",
    "CGATCTCATATGTTCCTGTTACAAAATTCGTACACCAAATTCTTGTTGTTTCGTTCGAATTTACCCATTT\n",
    "CACACACGAAAATCACATCCAAACGCAAAAATTCGTATCAGAAACTTATTATGAATGAGGAGGTAACCGG\n",
    "GTAGCAGATCTAATAATAGGCAACCAATCATACACTAAGACGTTTTGTGTGTTGCCTATTCAATTTTACT\n",
    "CCTGCTCCTGATATACCTCCTCGATCTCATATGTTCCTGTTACAAAATTCGTACACCAAATTCTTGTTGT\n",
    "TTCGTTCGAATTTACTCATTTCACACACGAAAAGCACATCTAAATGCAAAAATTCGTATCAGGAACATAT\n",
    "GATGAGTGAGGAGGTATCCGGGTAGCAGATCTAATATAGGCAACCAATCATACATTAAGACGTTTTGTGT\n",
    "GTTGCCTATTCAATTTTACTCCTGCTCCAGATATACCTCCTCGGTCTCTTTTATGTTCAAATGATTAACC\n",
    "TTGGAATCTTCACGTTTTACTAGTTTTTATGAGCAATTAGGAGGAGACGTATTTAAACACGAAATTCTGC\n",
    "TAAATTACCAGGTGAATCAATATGTAGCTCCAGACAACACTTAACCTTTTGTAGTTTAGGTCTTTCATCT\n",
    "TGCCTATATTTTTATTCCTATTTATTTCATTGTAAGATTTTATTCAGTAGGTCAATTAATGTATATAGTT\n",
    "TTTTTTCATGAAAATTGAATGATATATAGTTCTGTATGTATTTGTAGTAAGTTATACAGGGTGGCCCACC\n",
    "TGAGACACGGATCTCATTTTGAGGGAGTAAATGAAGATATTTAAAAAAAAATGTTTGTGATGTGTATAGG\n",
    "GTAGTTAGAAGCACAATTTAAAAATATATTCATATATACAGGGTGTTCGAAAACAAAGACATAGGCCAAA\n",
    "GATTTTGAAAATAACTTCATTTACTGCCTCAAAACGAGATCCGGATCTGGGGTGGGCCACCCAGTCTACT\n",
    "TGTAGATAAATTTATCAATAAACAAATAATTTAAATCTCACGTATTTAGGATGAAGAAAAAACAAAATTT\n",
    "AAAAACAATAAACAACATCTTTATTCTCATCTTCGAACAAATACAAAATGCCTTTTATAATACAGGGTAG\n",
    "TTATTACATGTGTCCAAATTTTCCTGTTTGGCCATTGCCAGCAAACAAAGGAAGTAATTCCTATCTTGCG\n",
    "GGCTCACAAAGAAGCTCAACACTGTCTTCGACTCGGCATCA\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Find 1:\n",
      "Pattern_1 (GATG) found at index 1302\n",
      "Pattern_2 (AGATAA) found at index 1310\n",
      "Number of nucleotides between the patterns: 4\n",
      "\n",
      "Find 2:\n",
      "Pattern_1 (GATG) found at index 1305\n",
      "Pattern_2 (AGATAA) found at index 1310\n",
      "Number of nucleotides between the patterns: 1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#WERK10 NinaB carotenoid \n",
    "#Gene ID: 123671684\n",
    "def find_patterns(genome_Seq, pattern_1, pattern_2):\n",
    "    distances = []\n",
    "    \n",
    "    \n",
    "    for i in range(len(genome_Seq) - len(pattern_1) + 1):\n",
    "        if genome_Seq[i:i+len(pattern_1)] == pattern_1:\n",
    "            \n",
    "            for j in range(i + len(pattern_1), min(i + len(pattern_1) + 20, len(genome_Seq) - 5)):  \n",
    "                current_subseq = genome_Seq[j:j+6]\n",
    "                if current_subseq in pattern_2:\n",
    "                    \n",
    "                    distance = j - (i + len(pattern_1))\n",
    "                    distances.append((i, j, distance, current_subseq))\n",
    "    \n",
    "    if not distances:\n",
    "        print(\"No occurrences of the patterns were found.\")\n",
    "    else:\n",
    "        for idx, (start_p1, start_p2, distance, found_pattern) in enumerate(distances):\n",
    "            print(f\"Find {idx + 1}:\")\n",
    "            print(f\"Pattern_1 ({pattern_1}) found at index {start_p1}\")\n",
    "            print(f\"Pattern_2 ({found_pattern}) found at index {start_p2}\")\n",
    "            print(f\"Number of nucleotides between the patterns: {distance}\\n\")\n",
    "\n",
    "\n",
    "genome_Seq = \"\"\"CGATGACCATTGGAATCTTTTTTCTACTATATTATTTTACTACCTAATTTTGCTATTAGGCGGGACTTTT\n",
    "GAATCACAATTATTCTGTGGATCTTTGCACTCTATGTGTATACTTACAGAACTACAAATTACGCGAATAT\n",
    "CTAGCCTGACTTTAGTTGCTTTTGGTGAGGTTTATTTCTAATTAAACATATTGAGTGAACCTTTCTCATG\n",
    "CTACGACCATAATTTTACAATAAAATTATTTATTCTCAATATGTACATAAATACGATCATAAATGATCAT\n",
    "ATCATCACAGTTGGCTTTGGGTTATCCCTTGCCTCTCACTACAAATTTACAATCCTTCATTATTTGTTAT\n",
    "GTAAACAAAATGTTTGTTCGTACACTTAAAAGTCCACATGCTCAGTTTAGAGAAAACTGAAACTTTTAAA\n",
    "TTGTACCATTTCATCTCAAAGACGAGCCTGCATAGGATTCCCATGTTGAAAAATTTTTACATAGTTCTCC\n",
    "TATCAATAGAATGATTTTAACTTCACTAATACAGCAAAATAGAATAAAACATTAAGTTCCATTTTTGAGA\n",
    "ATTTATTTAATCTAATTTTTCAGATATGTTCAAAAGGCCTGCACAAGACTCACTTGCTAATCTCCCCAAA\n",
    "AAAGTAAAGGAAGAACGAGAAAATTTGCAGTTGAATCCCTTTTTCAACATGAAATCAAGTATAAGAAGTG\n",
    "GAAAAGAATCAACATCAAAAGAGAATATTGATTTGGACATGTCAAATAATCCGTTTGGACAAATTCAATT\n",
    "GAAAAGAGCCATTAAAAATGAGACTGCAACAGACAATTTTGTCATAGCAAATGATAGAAATTCAGACTCC\n",
    "AATAATGATAATTCAAAAAAGAGGAAATGTGAAGACATTTTAGATTCTGTAATTTCAGAAAAACGTACAA\n",
    "AAATTATTTGTAAGAAAAATCGCAAATTTGTAAGAGTAGAATCTATTGATTTTTCAAATATGAATCATTC\n",
    "AGTCTCAAGTGGATCTTGGTTTGATAAGACCATCAAGACAGATATAAAACCAACACTTGAAAATTTTGAT\n",
    "GAGAATGAACAGAAATGGTGTGAAAGATTTAACAATTCTGTTGTTATTGTGGCAAAAAATTTGTGGAGAA\n",
    "AAGATATGTCAAATTCCAGTATCTCATCTAATTTAACTCATCGAACCAACGAGAATAATTTCAAAAAATT\n",
    "TCGAAAAGTGGCTGTGCCAAAGACGGATCATATTATTAAATGTTCGAAAATTGTCTTTGGGAAATCAAAT\n",
    "CTGGGAATTGACTACCAATATTCAGATGATGAAGATAATAGAAATATGTGAACTTTGTTTTTATATTTAT\n",
    "TATAAATGTGTATTATATTTATATTAATGGTAACTCATTAAATATTTTCTTGATGTTTTTCAACATTTTA\n",
    "TTGTACTCAGTTTCCAATGGTTTCAGAGCTCTATATCTGTATTTCCATTGAATCAAATCGCTTGATATAT\n",
    "AGTTGAATGATTAAGAATCATCCTTCGATTATAGTTCACAATTGAAAATGAGAAGGAATTACAAACTATG\n",
    "GGGCAAGATTGAATTTTTTAATGGGGTATCTGTATGTTTATTAGTTTACATACAAGCTTGGACATATATC\n",
    "GGTCAAGCTTTAAATTAATGATTAAATGATCAGATCCTTATATTCTGAATATTTTATTATTTGTTTTTCA\n",
    "TTAAAGAATTGCTTCACTTTGAACAAAATATTGATAAAAAAATTCCTTCAGCAGTGGGATAATACGCGTT\n",
    "TTATCTTCTCAAAGTTGAATAGCTTCATTTAAAGGTTTATCCAGTCCATAAATCGAAGCGACCTTCAATT\n",
    "TATTGAAGGTTAGGCGGATATATATCCTTGAAAATTTAATTCTATAATGGGTAAATATTGAATTATAAAG\n",
    "TTGCCTCGCGTATAAAACCTGCTCAACTGAATAAAATACCTGAATTACCGAAGCCCGTAAACGTCTGTTA\n",
    "CGTAATTTCAATTTTCATCCCATTTTCCGACCACTCGAATT\"\"\"  \n",
    "pattern_1 = \"GATG\"\n",
    "pattern_2 = [\"AGATAG\", \"TGATAG\", \"TGATAA\", \"AGATAA\"]\n",
    "\n",
    "\n",
    "find_patterns(genome_Seq, pattern_1, pattern_2)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
